### PR TITLE
Classic Templates

### LINKED ISSUE #1562: Classic Templates
TODO after classic release:

Remove code that merge spells from retail specs
Add these spells to the "classic" section

Needs to be done before next xpac

### PR DESCRIPTION
**Pull request by mrbuds**
# Description
Done
- Priest
- Mage
- Hunter
- Warlock
- Rogue
- Shaman
- Warrior
- Paladin
- Druid

Fixes #1562

### CODE REVIEW

**Comment by mrbuds** [comment]
Most abilities were added
Some classes could use more love for their buffs & debuffs categories
The "form" property is not used to make a stance/form trigger; instead a "usable = true" property is added to make it "blue" with conditions if in wrong form. This is not ideal ...
I don't have a druid high level enough on classic to get the right numbers for druid's forms, so i used 1 for cat, and {2, 3} for bear forms, these are placeholder

File: WeakAurasTemplates/TriggerTemplatesData.lua [hunk 1]
```diff
@@ -5051,32 +5586,40 @@ for regionType, regionData in pairs(WeakAuras.regionOptions) do
...
-              args[itemIndex] = item
-            end
-          end
-        end
-      end
+-- remove from tables data for wrong version of the game
+for classId, classData in pairs(templates.class) do
+  for id, data in pairs(classData) do
+    if (tonumber(id) and WeakAuras.IsClassic())
+    or (id == "classic" and not WeakAuras.IsClassic())
```
Comment (InfusOnWoW) [inline WeakAurasTemplates/TriggerTemplatesData.lua]: The structure of the templates looks hacky. As there's a classic table, but not a retail table. And those sections only exist in the class tables

Can you explain why you did it that way?

Comment (mrbuds) [reply WeakAurasTemplates/TriggerTemplatesData.lua]: I wanted to make it easier to re-use existing functions, but i'm re-organizing it to make it less hacky

File: WeakAurasTemplates/TriggerTemplates.lua [hunk 2]
```diff
@@ -824,63 +883,28 @@ local function subTypesFor(item, regionType)
...
       title = L["Show Only if Buffed"],
       description = L["Only shows the aura if the target has the buff."],
       createTriggers = function(triggers, item)
         createBuffTrigger(triggers, 1, item, "showOnActive", true);
       end,
-      data = { inverse = false },
+      data = data,
     });
+    if subglowindex then
+      data.subRegions = { [subglowindex] = subglow }
```
Comment (InfusOnWoW) [inline WeakAurasTemplates/TriggerTemplates.lua]: I don't understand this. Isn't the only difference between data and dataGlow that dataGlow has the glow sub region?
So essentially, aren't after this data and dataGlow identically?

File: WeakAurasTemplates/TriggerTemplatesData.lua [hunk 3]
```diff
@@ -121,4243 +119,4898 @@ local pvpAzeriteTraits = {
...
-        { spell = 287731, type = "buff", unit = "player"}, --Empyreal Ward
-      },
-      icon = 135349
-    },
-    [6] = {},
-    [7] = {
-      title = L["PvP Talents"],
-      args = {
-        { spell = 236186, type="ability", pvptalent = 4},-- Cleansing Light
-        { spell = 247675, type="ability
```
Comment (InfusOnWoW) [inline WeakAurasTemplates/TriggerTemplatesData.lua]: Minor nitpick: A comment here that retail follows would be nice

File: WeakAurasTemplates/TriggerTemplatesData.lua [hunk 4]
```diff
@@ -4616,12 +5268,144 @@ end
...
+  for i = 1, 3 do
+    tinsert(templates.class.DEATHKNIGHT[i][8].args, createSimplePowerTemplate(6));
+
+    tinsert(templates.class.DEATHKNIGHT[i][8].args, {
+      title = L["Runes"],
+      icon = "Interface\\Icons\\spell_deathknight_frozenruneweapon",
+      triggers = {[1] = { trigger = { type = "status", event = "Death Knight Rune", unevent = "auto"}}}
+    });
+  end
+else
```
Comment (InfusOnWoW) [inline WeakAurasTemplates/TriggerTemplatesData.lua]: Same here, the if is 100 lines above.

File: WeakAurasTemplates/TriggerTemplates.lua [hunk 5]
```diff
@@ -904,17 +928,19 @@ local function subTypesFor(item, regionType)
       createConditions = function(conditions, item, regionType)
         missingBuffGreyed(conditions, 1, regionType);
       end,
-      data = { inverse = false },
+      data = data,
```
Comment (InfusOnWoW) [inline WeakAurasTemplates/TriggerTemplates.lua]: Haven't tested this exact code path "Always shows the aura, grey if buff not active." does indeed include a unused glow sub region. So I believe the above code is a left-over?

**Pull request merged**